xonsh/jobs.py

```diff
@@ -87,18 +87,39 @@ def _set_pgrp(info):
 
     _shell_pgrp = os.getpgrp()
 
-    _block_when_giving = (signal.SIGTTOU, signal.SIGTTIN, signal.SIGTSTP)
-
-    def _give_terminal_to(pgid):
-        # over-simplified version of:
-        #    give_terminal_to from bash 4.3 source, jobs.c, line 4030
-        # this will give the terminal to the process group pgid
-        if _shell_tty is not None and os.isatty(_shell_tty):
-            if not ON_CYGWIN:
+    _block_when_giving = (signal.SIGTTOU, signal.SIGTTIN,
+                          signal.SIGTSTP, signal.SIGCHLD)
+
+    # _give_terminal_to is a simplified version of:
+    #    give_terminal_to from bash 4.3 source, jobs.c, line 4030
+    # this will give the terminal to the process group pgid
+    if ON_CYGWIN:
+        import ctypes
+        _libc = ctypes.CDLL('cygwin1.dll')
```

Comment:
>  since libc is something else

Personally, I don't think it is.  I think this is _one implementation_ of libc.  From the libc man page:

> The term "libc" is commonly used as a shorthand for the "standard C library", a library of standard functions that can be used by all C programs (and sometimes by programs in other languages).

This just happens to be Cygwin's implementation of libc (as opposed to `glibc`, for example).  It has all the same stuff as if we had done `ctypes.CDLL('libc.so.6')` under GNU/Linux).  So I would argue that it's just as much libc as the one on GNU/Linux would be (in fact, one could imagine an implementation that used `ctypes` for both Cygwin and normal operation, but just had:

```
if ON_CYGWIN:
    _libc = ctypes.CDLL('cygwin1.dll')
elif ON_DARWIN:
    _libc = ctypes.CDLL('libSystem.B.dylib')
else:
    _libc = ctypes.CDLL('libc.so.6')
```

That said, if it speeds this one on its way, I'm happy to change the name.  Just let me know.